rules/CodingStyle/Rector/Property/InlineSimplePropertyAnnotationRector.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Property;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassConst;
+use PhpParser\Node\Stmt\Property;
+use Rector\Core\Contract\Rector\AllowEmptyConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Tests\CodingStyle\Rector\Property\InlineSimplePropertyAnnotationRector\InlineSimplePropertyAnnotationRectorTest
+ */
+final class InlineSimplePropertyAnnotationRector extends AbstractRector implements AllowEmptyConfigurableRectorInterface
+{
+    /**
+     * @var string[]
+     */
+    private array $annotationsToConsiderForInlining = ['@var', '@phpstan-var', '@psalm-var'];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Inline simple @var annotations (or other annotations) when they are the only thing in the phpdoc',
+            [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /**
+     * @phpstan-var string
+     */
+    private const TEXT = 'text';
+
+    /**
+     * @var DateTime[]
+     */
+    private ?array $dateTimes;
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    /** @phpstan-var string */
+    private const TEXT = 'text';
+
+    /** @var DateTime[]|null */
+    private ?array $dateTimes;
+}
+CODE_SAMPLE
+,
+                    ['var', 'phpstan-var'],
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class, ClassConst::class];
+    }
+
+    /**
+     * @param mixed[] $configuration
+     */
+    public function configure(array $configuration): void
+    {
+        Assert::allString($configuration);
+
+        $this->annotationsToConsiderForInlining = array_map(
+            fn (string $annotation): string => '@' . $annotation,
```

Comment:
`ltrim` can be used if user add `@` prefix:

```suggestion
            fn (string $annotation): string => '@' . ltrim($annotation, '@')
```

Reply:
Good idea. I actually did not prepepend the `@` initially, but I found out that it breaks the docs generation, because the string is then interpreted as service reference. :slightly_smiling_face: